### ISSUE TITLE
Bug: `no-useless-escape` incorrectly reports `[\/\\]` cases

### ISSUE THREAD

**Issue by dimaMachina** [eslint/eslint#19126]
### Environment

Node version:
npm version:
Local ESLint version:
Global ESLint version:
Operating System:

### What parser are you using?

Default (Espree)

### What did you do?

https://eslint.org/play/#eyJ0ZXh0IjoiL1tcXC9cXFxcXVxcZCtfZG9jdW1lbnQuZ3JhcGhxbCQvIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby11c2VsZXNzLWVzY2FwZSI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### What did you expect to happen?

no errors as it became an invalid pattern

Remove the `\`. This maintains the current functionality.

# with
![Image](https://github.com/user-attachments/assets/7ad9cfb4-4c86-4c91-9b93-56c288e03365)

# without
![Image](https://github.com/user-attachments/assets/0cfd19e0-035d-4ec3-b9c5-bc19bf658afb)

### What actually happened?

Error: Remove the `\`. This maintains the current functionality.

which is wrong

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiL1tcXC9cXFxcXVxcZCtfZG9jdW1lbnQuZ3JhcGhxbCQvIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby11c2VsZXNzLWVzY2FwZSI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
This works as intended because `/` doesn't need to be escaped inside `[]`.

```js
/[/\\]\d+_document.graphql$/.test("/1_document.graphql"); // true
```

(if you're using regex101, switch Flavor in the left panel to `ECMAScript (JavaScript)`)

**Comment by dimaMachina** [comment]
Thank you, sorry for noise
